server/service/src/document/Programs.md

```diff
@@ -0,0 +1,144 @@
+A medical `program` focuses on the treatment of a medical condition and a patient can be enrolled in one or more programs.
+
+All program related data is recorded in `documents` which resemble physical forms.
+For example, there are currently three main document types:
+
+1. The `Patient` document records general patient information such as name, address or age.
+   The Patient document is shared between all programs of a patient.
+2. The `Program Enrolment` document holds patient specific information about a single program the patient is enrolled in.
+3. `Program Encounter` documents record data about a patients visit at a doctor.
+   Each program encounter is associate with a single program and there is an encounter document for every visit.
+
+What data is stored in the documents is widely customizable and its possible to add new custom document types.
+This makes the system customizable to fullfil the needs of different clients.
+For example, what data is recorded or how data is displayed and edited can be defined by the client.
+
+## Practitioner driven data recording
+
+In general data recording is driven by the practitioner.
+For example, the practitioner has to explicitly change a patient's data rather than the patient data is indirectly updated based on some defined logic.
+This also means that the practitioner is responsible for inputting data correctly and no inadvertently side effects interfere with the practitioner's intentions.
+
+## Program Events
+
+While practitioner are the main driver for entering data there are certain use-cases where data is set in a predefined way.
+Currently this is done using program `events`.
+Events are stored in the document but are filled automatically, e.g. by the ui form controls while editing a form.
+For example, an encounter is labeled depending on the entered data.
+Events are currently used for the following purposes:
+
+1. Based on the dispensed pill count encounter two events are scheduled in the future to change the status to "treatment interrupted" or to "lost"
+2. Based on encounter fields set or not set, an encounter is labeled "Pending Lab Report" or "Lap Report Received"
+
+Program events have a context field which described where events belong to, e.g. events can be scheduled from a DemoEncounter but be in the DemoProgram context.
+Moreover, within a context events can be grouped.
+This is useful when updating events since it makes it easier for the a UI control to replace the correct events.
+
+When updating a document (currently only encounters documents) the backend extract events from the document and put these events it into a `program_event` table.
+Events from previous encounter document versions are removed or replaced in this process.
+This table can, for example, be queried to find the currently active encounter or program status, i.e. querying the status events that are not scheduled in the future.
```

Comment:
```suggestion
When updating a document (currently only encounters documents) the backend extracts events from the document and put these events it into a `program_event` table.
Events from previous encounter document versions are removed or replaced in this process.
This table can, for example, be queried to find the current active encounter or program status, or be queried to find status events that are not scheduled in the future
```

Reply:
thanks now:

When updating a document (currently only encounter documents) the backend extracts events from the document and put these events into a `program_event` table.
Events from previous encounter document versions are removed or replaced in this process.
This table can, for example, be queried to find the currently active encounter or program status, i.e. queried to find the status events that are not scheduled in the future.

Reply:
current active encounter

i.e means that is... but the statement before and after i.e are stating different things... or am I missing something? 😭

Reply:
Ok thanks, how about?

This table can, for example, be used to find the current encounter status by querying the latest status event which is not scheduled in the future.

Reply:
👍